File: 20221102/모범답안/20221102_02/CountMatchedNumber_test.js
import assert from 'assert'
import solution from './CountMatchedNumber.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const s = '134246'
    const n = 4

    // when
    const result = solution(s, n)

    // then
    assert.deepStrictEqual(2, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const s = '4882305038'
    const n = 8

    // when
    const result = solution(s, n)

    // then
    assert.deepStrictEqual(3, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const s = '0'
    const n = 0

    // when
    const result = solution(s, n)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const s = '0'
    const n = 1

    // when
    const result = solution(s, n)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const s = '44435013785452732466'
    const n = 4

    // when
    const result = solution(s, n)

    // then
    assert.deepStrictEqual(5, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const s = '30812522681139093477'
    const n = 7

    // when
    const result = solution(s, n)

    // then
    assert.deepStrictEqual(2, result)
    done();
  });
})

File: 20221102/모범답안/20221102_02/CountMatchedNumber_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.easy.CountMatchedNumber.CountMatchedNumber import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        s = '134246'
        n = 4

        # when
        result = solution(s, n)

        # then
        self.assertEqual(2, result)

    def test_2(self):
        # given
        s = '4882305038'
        n = 8

        # when
        result = solution(s, n)

        # then
        self.assertEqual(3, result)

    def test_3(self):
        # given
        s = '0'
        n = 0

        # when
        result = solution(s, n)

        # then
        self.assertEqual(1, result)

    def test_4(self):
        # given
        s = '0'
        n = 1

        # when
        result = solution(s, n)

        # then
        self.assertEqual(0, result)

    def test_5(self):
        # given
        s = '44435013785452732466'
        n = 4

        # when
        result = solution(s, n)

        # then
        self.assertEqual(5, result)

    def test_6(self):
        # given
        s = '30812522681139093477'
        n = 7

        # when
        result = solution(s, n)

        # then
        self.assertEqual(2, result)


if __name__ == '__main__':
    unittest.main()
